### ISSUE TITLE
Wrong redshift of absorber in fast_metal_dmat

### ISSUE THREAD

**Issue by cramirezpe** [igmhub/picca#1044]
There's an issue in the new picca_fast_metal_dmat that is causing the redshift of the absorbers to be at a smaller redshift than it should.

Calum just shared this plot:
![image](https://github.com/igmhub/picca/assets/58682644/478b1633-5d1b-4c12-bc43-2b1237db7580)
The difference in number of points is probably due to the use of higher resolution in the fast one.

We still need to check the xdmat case.

**Comment by andreufont** [comment]
@calumgordon - could you check what happens with the ~500 bins in the blue histogram that have z_eff_civ=0?

**Comment by andreufont** [comment]
About the total number of point in that histogram, shouldn't these matrices have 2500^2 elements, i.e., (50x50)^2? This would be 6250000 (>6M) but I don't see that many points in the histogram...

**Comment by calumgordon** [comment]
It's the CIVxCIV correlation in the metal dmat, which has 150x100 bins in the fast case, and 75x50 in the slow. 

For the bins with z_eff = 0, there isn't a lot of info easily available, so I'm not sure.

**Comment by calumgordon** [comment]
Ok, here is a plot of RT_CIV, where we can see the missing bins from 176 to 200 Mpc/h. I've also checked and this indeed corresponds to (200-176)/4  *  75 = 450 bins. 


![RT_CIV](https://github.com/igmhub/picca/assets/57399293/4e35c2f7-70a8-4dd8-832e-8bfcdc1ffffa)

**Comment by andreufont** [comment]
This is soooo confusing. I first thought that it would be a few rt bins that happens to not be covered because of the 0.1% of data used in the computation of matrices, but the fact that they are all localised at rt > 175 Mpc/h means that this is more likely a bug somewhere... It would be quite a coincidence.

My next guess is that somewhere in the code, one uses the CIV redshift (and not the Lya) to compute maximum angular separation to include in the search for neighboring quasars, and that this cause quasars at > 175 Mpc/h (in Lya coordinates) to not be included because they correspond to > 200 Mpc/h in SiIV coordinates

**Comment by calumgordon** [comment]
RT_LYA - RT_CIV vs RT_LYA. There is a difference between the co-ordinates as expected but it's quite small.
![rt_diff_lya_civ](https://github.com/igmhub/picca/assets/57399293/f99d166d-c966-4776-88f0-3cde7b1bc126)

**Comment by andreufont** [comment]
I don't think this makes sense, but at this point I have given up on understanding how we the old code works. 

I need to go back to the blackboard and write down a simple example to make sure I understand how it works.

**Comment by andreufont** [comment]
@calumgordon - have looked at this any further? It would be good to (eventually) understand these values (for both the old and new metal implementation)

**Comment by andreufont** [comment]
@calumgordon , @julienguy - was this fixed / understood? If so, could we close the issue?
